NomaiGrandPrix/NomaiGrandPrix.cs

```diff
@@ -271,13 +271,46 @@ private void InitMapMarker()
 
             var labelText = $"GOAL: {SpeedrunState.GoalPoint?.displayName.ToUpper()}";
             var markerManager = Locator.GetMarkerManager();
+
+            OuterFogWarpVolume outerWarp = null;
+            if (SpeedrunState.GoalPoint.Value.internalId.Equals("SpawnPoint_ElsinoreSeed"))
+            {
+                outerWarp = GameObject.Find("DB_Elsinore_Body/Sector_ElsinoreDimension/Interactables_ElsinoreDimension/OuterWarp_Elsinore")
+                    .GetComponent<OuterFogWarpVolume>();
+            }
+            else if (SpeedrunState.GoalPoint.Value.internalId.Equals("SpawnPoint_EscapePod"))
+            {
+                outerWarp = GameObject.Find("DB_EscapePodDimension_Body/Sector_EscapePodDimension/Interactables_EscapePodDimension/OuterWarp_EscapePod")
+                    .GetComponent<OuterFogWarpVolume>();
+            }
+            else if (SpeedrunState.GoalPoint.Value.internalId.Equals("SpawnPoint_PioneerCamp"))
+            {
+                outerWarp = GameObject.Find("DB_PioneerDimension_Body/Sector_PioneerDimension/Interactables_PioneerDimension/OuterWarp_Pioneer")
+                    .GetComponent<OuterFogWarpVolume>();
+            }
+            else if (SpeedrunState.GoalPoint.Value.internalId.Equals("SpawnPoint_NomaiGrave"))
+            {
+                outerWarp = GameObject.Find("DB_EscapePodDimension_Body/Sector_EscapePodDimension/Interactables_EscapePodDimension/OuterWarp_EscapePod")
+                    .GetComponent<OuterFogWarpVolume>();
+            }
+            else if (SpeedrunState.GoalPoint.Value.internalId.Equals("Spawn_Vessel"))
+            {
+                outerWarp = GameObject.Find("DB_VesselDimension_Body/Sector_VesselDimension/Interactables_VesselDimension/OuterWarp_Vessel")
+                    .GetComponent<OuterFogWarpVolume>();
+            }
+
```

Comment:
wdyt about moving the goal point and game object path mappings into a Dictionary to reduce duplication?
```csharp
// Put this in a const outside the method def:
private static Dictionary<string, string> GameObjectPathsBySpawnPointID = new Dictionary<string, string>
{
    ["SpawnPoint_ElsinoreSeed"] = "DB_Elsinore_Body/Sector_ElsinoreDimension/Interactables_ElsinoreDimension/OuterWarp_Elsinore",
    ["SpawnPoint_EscapePod"] = "DB_EscapePodDimension_Body/Sector_EscapePodDimension/Interactables_EscapePodDimension/OuterWarp_EscapePod",
    ["SpawnPoint_PioneerCamp"] = "DB_PioneerDimension_Body/Sector_PioneerDimension/Interactables_PioneerDimension/OuterWarp_Pioneer",
    ["SpawnPoint_NomaiGrave"] = "DB_EscapePodDimension_Body/Sector_EscapePodDimension/Interactables_EscapePodDimension/OuterWarp_EscapePod",
    ["Spawn_Vessel"] = "DB_VesselDimension_Body/Sector_VesselDimension/Interactables_VesselDimension/OuterWarp_Vessel"
};

// Replace the if-else chain with:
OuterFogWarpVolume outerWarp = null;
var spawnPointId = SpeedrunState.GoalPoint.Value.internalId;

string gameObjectPath;
if (GameObjectPathsBySpawnPointID.TryGetValue(spawnPointId, out gameObjectPath)) {
    outerWarp = GameObject.find(gameObjectPath).GetComponent<OuterFogWarpVolume>();
}
```

Reply:
I forgot to say that this is non-blocking, sorry!

Reply:
fixed!